src/Azure.Functions.Cli/Helpers/ExtensionsHelper.cs

```diff
@@ -63,7 +63,12 @@ public static IEnumerable<ExtensionPackage> GetExtensionPackages()
                 }
             }
 
-            packages.Add("ExtensionsMetadataGeneratorPackage", Constants.ExtensionsMetadataGeneratorPackage);
+            // We only need extensionsMetadatGeneratorPackage, if there is a trigger type that requires an extension
+            // So, if we didn't find any extension packages, we don't need to add this.
+            if (packages.Count != 0)
```

Comment:
This would break the logic here. since the package count will be different. need to update that as well.

https://github.com/Azure/azure-functions-core-tools/blob/dev/src/Azure.Functions.Cli/ExtensionBundle/JsonFileConfigurationSource.cs#L71-L73

Reply:
That checks for less than equals to 1, and removes extensionsmeta if it is present. I don't think the logic there should break. But I can update that to 0 I suppose?